app/states/fantasyState.js

```diff
@@ -74,12 +75,22 @@ const fantasyState = {
   enemyPathFinding(enemyKey) {
     const enemy = localState.enemies[enemyKey]
     StackQuest.game.physics.arcade.overlap(projectile.bullets, enemy, () => {
-      graveyard.push(enemy)
-      delete localState.enemies[enemyKey]
+      // console.log('enemy in fantasy state is:', enemy)
+      //make sure projectile stops AS SOON AS it hits target
+      let didDie = enemy.takeDamage(projectile.damage)
+      if (didDie) {
+        graveyard.push(enemy)
+        delete localState.enemies[enemyKey]
+      }
     })
     StackQuest.game.physics.arcade.overlap(enemy, playerObject, () => {
-      playerObject.position.x = 200
-      playerObject.position.y = 200
+      playerObject.internalStats.hp -= enemy.attack()
```

Comment:
@gabrielwr for future refactoring efforts put lines 86-94 in another function (as it is repeated in spaceState.js in lines 85-93)